cmd/main.go

```diff
@@ -388,6 +395,8 @@ func main() {
 		apiEnabler.Enable()
 	}
 
+	go setupKubeAPI(log, Options.EnableKubeAPI)
```

Comment:
why do we need the manager/controllers setup in the goroutine? the only thing that needs really needs to run in go-routine is mgr.Start(). moving mgr/controller setup to the main will allow us to use common variables, like mgr.Client etc'

Reply:
Maybe it should run in parallel anyway? this flow does not affect the rest-api flow and there is no reason to block the startup when running `SetupWithManager`.

Reply:
i think that running setups in parallel is a little bit overkill: all those functions are not time consuming, so they are not really blocking Rest API. What happens if manager/controller setup fails? Don't we want to fails the whole process? Also, you will still need to make sure that manager creation runs before client creation of bminventory ( for cache sake)

Reply:
Manager is the first thing that is being created.
If manager or controller setup fails then all the service fail to start.
what do you mean by 
> you will still need to make sure that manager creation runs before client creation of bminventory 
I need inventory api and other interfaces in order to init the controller, they will call those interfaces.